src/github.com/getlantern/connpool/connpool.go

```diff
@@ -46,11 +49,12 @@ type Pool struct {
 	runMutex   sync.Mutex
 	running    bool
 	actualSize int
```

Comment:
`actualSize` is assigned as `p.Size` when `Start()` but never changes afterwards.

In fact, no matter how large wg is in `Stop()`, [L153](https://github.com/getlantern/flashlight-build/blob/peerscanner/src/github.com/getlantern/connpool/connpool.go#L153) will `wg.Done()` it.